selfdrive/car/honda/carstate.py

```diff
@@ -260,18 +259,20 @@ def update(self, cp, cp_cam, cp_body):
     else:
       ret.cruiseState.speed = cp.vl["CRUISE"]["CRUISE_SPEED_PCM"] * CV.KPH_TO_MS
 
-    self.brake_switch = cp.vl["POWERTRAIN_DATA"]["BRAKE_SWITCH"] != 0
     if self.CP.carFingerprint in HONDA_BOSCH_ALT_BRAKE_SIGNAL:
       ret.brakePressed = cp.vl["BRAKE_MODULE"]["BRAKE_PRESSED"] != 0
     else:
       # brake switch has shown some single time step noise, so only considered when
       # switch is on for at least 2 consecutive CAN samples
-      # panda safety only checks BRAKE_PRESSED signal
-      ret.brakePressed = bool(cp.vl["POWERTRAIN_DATA"]["BRAKE_PRESSED"] or
-                              (self.brake_switch and self.brake_switch_prev and cp.ts["POWERTRAIN_DATA"]["BRAKE_SWITCH"] != self.brake_switch_prev_ts))
-
-      self.brake_switch_prev = self.brake_switch
-      self.brake_switch_prev_ts = cp.ts["POWERTRAIN_DATA"]["BRAKE_SWITCH"]
+      # brake switch rises earlier than brake pressed but is never 1 when in park
+      brake_switch_vals = cp.updated["POWERTRAIN_DATA"]["BRAKE_SWITCH"]
+      if len(brake_switch_vals):
+        if cp.vl["POWERTRAIN_DATA"]["BRAKE_SWITCH"] != 0:
+          # use all updated values this cycle, sometimes we receive two messages at once
+          self.brake_switch_on += sum(brake_switch_vals)
```

Comment:
We increment the `self.brake_switch_on` variable if the message is updated this frame, and the last value isn't 0. So if we get two 1's, we increment twice and set `brakePressed` to true. If we get one or more 1's and one 0, then we reset it to 0.